Resources/Prototypes/Entities/Structures/Machines/telecomms.yml

```
@@ -8,23 +8,138 @@
     sprite: Structures/Machines/telecomms.rsi
     snapCardinals: true
     layers:
-    - state: icon
-    - state: unlit
+    - state: tele-server-nopower
+    - state: tele-stripe
+    - state: tele-server-on
       shader: unshaded
-      map: ["enum.PowerDeviceVisualLayers.Powered"]
-    - state: panel
+      visible: false
+      map: ["tele-server-on"]
+    - state: tele-server-no-key
+      map: ["tele-server-no-key"]
+    - state: tele-server-on-common
+      map: ["tele-server-on-common"]
+    - state: tele-server-on-cargo
+      map: ["tele-server-on-cargo"]
+    - state: tele-server-on-engineering
+      map: ["tele-server-on-engineering"]
+    - state: tele-server-on-medical
+      map: ["tele-server-on-medical"]
+    - state: tele-server-on-science
+      map: ["tele-server-on-science"]
+    - state: tele-server-on-sec
+      map: ["tele-server-on-sec"]
+    - state: tele-server-on-service
+      map: ["tele-server-on-service"]
+    - state: tele-server-on-command
+      map: ["tele-server-on-command"]
+    - state: tele-server-on-multiple
+      map: ["tele-server-on-multiple"]
+    - state: tele-server-on-default
+      map: ["tele-server-on-default"]
+    - state: tele_o
       map: ["enum.WiresVisualLayers.MaintenancePanel"]
+  - type: ItemMapper
+    sprite: Structures/Machines/telecomms.rsi
+    mapLayers:
+      tele-server-on-common:
+        whitelist:
+          tags:
+          - EncryptionCommon
+      tele-server-on-cargo:
+        whitelist:
+          tags:
+          - EncryptionCargo
+      tele-server-on-engineering:
+        whitelist:
+          tags:
+          - EncryptionEngineering
+      tele-server-on-medical:
+        whitelist:
+          tags:
+          - EncryptionMedical
+      tele-server-on-science:
+        whitelist:
+          tags:
+          - EncryptionScience
+      tele-server-on-sec:
+        whitelist:
+          tags:
+          - EncryptionSecurity
+      tele-server-on-service:
+        whitelist:
+          tags:
+          - EncryptionService
+      tele-server-on-command:
+        whitelist:
+          tags:
+          - EncryptionCommand
+      tele-server-on-default:
```

If you move this layer before the departments in `SpriteComponent`, you can do away with the `EncryptionElse` tag. I'm scared by the reliance on layer order but it works and I don't know if I'm supposed to be scared, anyway.
```suggestion
      tele-server-on-default:
        whitelist:
          components:
          - EncryptionKey
        minCount: 1
```

---

Agreed, better implimentation